documentation recommends accessing `config.cache` in an unsafe way

https://docs.pytest.org/en/stable/how-to/cache.html#the-new-config-cache-object

in this example recommends accessing `pytestconfig.cache` which will crash at runtime if the cacheprovider plugin has been disabled as mentioned [here](https://docs.pytest.org/en/stable/how-to/cache.html#usage):

```
_______________________________________________________________________________ ERROR at setup of test_function ________________________________________________________________________________

pytestconfig = <_pytest.config.Config object at 0x749189eaa7b0>

    @pytest.fixture
    def mydata(pytestconfig):
>       val = pytestconfig.cache.get("example/value", None)
E       AttributeError: 'Config' object has no attribute 'cache'

python/tests/test_asdf.py:12: AttributeError
=================================================================================== short test summary info ====================================================================================
ERROR python/tests/test_asdf.py::test_function - AttributeError: 'Config' object has no attribute 'cache'
======================================================================================= 1 error in 0.07s =======================================================================================
```

pytest should set the `cache` object to `None` when the plugin isn't enabled. that way, type checkers will report an error when attempting to use it without checking whether it's `None`

--- a/changelog/14148.doc.rst
+++ b/changelog/14148.doc.rst
@@ -1,0 +1,2 @@
+Documented a safe ``pytestconfig.cache`` access pattern when the
+``cacheprovider`` plugin is disabled.

--- a/doc/en/how-to/cache.rst
+++ b/doc/en/how-to/cache.rst
@@ -200,7 +200,8 @@ The new config.cache object
 Plugins or conftest.py support code can get a cached value using the
 pytest ``config`` object.  Here is a basic example plugin which
 implements a :ref:`fixture <fixture>` which reuses previously created state
-across pytest invocations:
+across pytest invocations. The example below also works when the
+``cacheprovider`` plugin is disabled:
 
 .. code-block:: python
 
@@ -214,11 +215,16 @@ across pytest invocations:
 
     @pytest.fixture
     def mydata(pytestconfig):
-        val = pytestconfig.cache.get("example/value", None)
+        cache = getattr(pytestconfig, "cache", None)
+        if cache is None:
+            expensive_computation()
+            return 42
+
+        val = cache.get("example/value", None)
         if val is None:
             expensive_computation()
             val = 42
-            pytestconfig.cache.set("example/value", val)
+            cache.set("example/value", val)
         return val
 
 

--- a/doc/en/how-to/cache.rst
+++ b/doc/en/how-to/cache.rst
@@ -200,8 +200,7 @@ The new config.cache object
 Plugins or conftest.py support code can get a cached value using the
 pytest ``config`` object.  Here is a basic example plugin which
 implements a :ref:`fixture <fixture>` which reuses previously created state
-across pytest invocations. The example below also works when the
-``cacheprovider`` plugin is disabled:
+across pytest invocations:
 
 .. code-block:: python
 
@@ -217,6 +216,8 @@ across pytest invocations. The example below also works when the
     def mydata(pytestconfig):
         cache = getattr(pytestconfig, "cache", None)
         if cache is None:
+            # pytestconfig not having the cache attribute means the
+            # cache plugin is disabled.
             expensive_computation()
             return 42
 

